### PR TITLE
Require Elixir v1.12 for new apps

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

**Comment by josevalim** [comment]
Hi @agbell, can you please give me a quick hand here? We need to do two changes:

1. The caching directory needs to be per Elixir version (if it isn't yet)

2. We should only run the installer tests on certain Elixir versions (in this case, v1.12) - how to split them and keep them efficiently?

Thanks!

**Comment by agbell** [comment]
Hi @josevalim,
Yep, I'll take a look at it tomorrow.

**Comment by agbell** [comment]
Check this out @josevalim https://github.com/phoenixframework/phoenix/pull/4348/files#diff-2705f6016a64189abd14041e2f3e8dbb829553295d9088e4eb42eccb5e61bf47R109

File: Earthfile [hunk 1]
```diff
@@ -15,13 +15,16 @@ test:
     # Run unit tests
     RUN mix test
 
-    # Run installer tests
-    WORKDIR /src/installer
-    RUN mix test
+    IF $RUN_INSTALLER_TESTS == "1"
```
Comment (josevalim) [inline Earthfile:18]:
```suggestion
    IF [ $RUN_INSTALLER_TESTS = "1" ]
```

File: Earthfile [hunk 2]
```diff
@@ -15,13 +15,16 @@ test:
     # Run unit tests
     RUN mix test
 
-    # Run installer tests
-    WORKDIR /src/installer
-    RUN mix test
+    IF [ $RUN_INSTALLER_TESTS = "1" ]
```
Comment (josevalim) [inline Earthfile:18]:
```suggestion
    IF [ "$RUN_INSTALLER_TESTS" = "1" ]
```

Comment (josevalim) [inline Earthfile]: @agbell please let me know if this is the way to go! Thank you so much! :heart:

File: Earthfile [hunk 3]
```diff
@@ -15,13 +15,16 @@ test:
     # Run unit tests
     RUN mix test
 
-    # Run installer tests
-    WORKDIR /src/installer
-    RUN mix test
+    IF [ "$RUN_INSTALLER_TESTS" = "1" ]
```
Comment (josevalim) [inline Earthfile:18]: @agbell hrm, this is not working, but it should according to the docs. Is this perhaps a Earthfile version issue?

Comment (agbell) [reply Earthfile:18]: It must be, its a newer feature.  in the GitHub action, it has the download step, we could bump it to 
https://github.com/earthly/earthly/releases/tag/v0.5.16

Comment (josevalim) [reply Earthfile:18]: It works!!!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**